app/src/main/java/com/youssef/fixit/Jobs/IAPIListener.java

```diff
@@ -0,0 +1,13 @@
+package com.youssef.fixit.Jobs;
+
+import com.youssef.fixit.Models.Jobs.Jobs;
+
+import retrofit2.Response;
+
+interface IAPIListener {
```

Comment:
We will need to keep this more generic. Since any presenter will implement it to get any data type result it's better to keep it generic.

It can be something like this:

```
IDataProviderListener<T> {
void onSuccess(T result);
void onError(Error error);
....
}
```

Then the presenter will be implementing it like this 

```
class JobsPresenter implements IAPIListener<List<Response>> { ... }
```


Like that, when you need to change the data source from network to local, you won't need to send `Response<?> anymore just list of your data or errors. 

Also, try to make data fetching errors more unified. You can create error types. something like this:

class Error { 
String message;
Int errorCode;
}

and it will be set from whether your network layer or database layer.

Reply:
Also for the naming convention try not to name the abbreviation all in capitals. Only the first letter should be in capital for a class name. so it should be like this `ApiListener`. Also no need for the `I` since listener is sufficient.